Posit Infrastructure/Guide to using renv to create reproducible projects.md

```diff
@@ -0,0 +1,47 @@
+# Guide to using [{renv}](https://rstudio.github.io/renv/) to create reproducible projects
+
+## Purpose
+
+Our code should be reproducible and shareable. But sometimes things fail to run properly when we return to them, or on a colleague's machine. The R package [{renv}](https://rstudio.github.io/renv/) ensures that the exact set of packages (and their versions) required to run the code in a specific project are recorded and kept available to current and future users. 
+
+This guide provides basic step-by-step instructions for using [{renv}](https://rstudio.github.io/renv/), as well as more detailed information. 
+
+## Step-by-step instructions
+
+Pre-requisite: a project (preferably version-controlled using git).
+
+#### Setting up [{renv}](https://rstudio.github.io/renv/) for the first time in a project:
+1.	Open the project.
+2.	Call `renv::init()` to initialise renv within that project. This can take a while depending on the size of the project.
+3.	Calling `renv::status()` should confirm the new 'lockfile' is synced with the project (i.e., contains details of all the required packages).
+4.	Commit the changes (additional files and folders) to version control.
+
+#### Adding (/removing) packages to the project:
+1.	Install (or remove) the required packages as usual. 
+2.	Change the code, by adding (or removing) calls to `library()` or `require()`.
+3.	Calling `renv::status()` will tell you there are changes that aren’t recorded in the lockfile.
+4.	Calling `renv::snapshot()` will record any changes into the lockfile.
+5.	Calling `renv::status()` should now confirm everything's synced. 
+6.	Commit the changes to version control.
+
+#### Using a project that has already had [{renv}](https://rstudio.github.io/renv/) initialised (e.g., when collaborating on a colleague's project):
+1.	Ensure the project is synced with the latest version-controlled repository (e.g., `git pull` from github).
+2.	Call `renv::restore()` to make sure your local version of the project has all the required packages.
+3.	Calling `renv::status()` should now confirm everything's synced. 
```

Comment:
We might want to include a note about packages that may no longer be available on a CRAN mirror, and how to resolve this issue.  Or perhaps that might scare people off.

Reply:
Hey, @terrymclaughlin! Do you have an example here? Presumably, the CRAN archive would contain any package that has and now no longer is available on CRAN and this would be handled?